src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageAssets.cs

```diff
@@ -1475,4 +1481,27 @@ private void ComputePackageExclusions()
             }
         }
     }
+
+    internal static class NuGetTargetFrameworkUtils
+    {
+        public static NuGetFramework GetTargetFramework(string targetFrameworkMoniker, string targetPlatformMoniker)
+        {
+            var targetFramework = NuGetUtils.ParseFrameworkName(targetFrameworkMoniker);
+            if (targetPlatformMoniker != null && !targetPlatformMoniker.Equals(string.Empty))
+            {
+                try
+                {
+                    var targetPlatform = NuGetUtils.ParseFrameworkName(targetPlatformMoniker);
```

Comment:
In https://github.com/NuGet/NuGet.Client/compare/dev-nkolev92-tfmImprovements?expand=1 I am adding a ParseComponents method that should take a TFM & TPM.